### PR TITLE
Fix LogErr when passed None as failure

### PR DESCRIPTION
**Pull request by radix**
When LogErr was passed None, it wouldn't properly get the exception from the context, since the exception context was lost by the time the intent was performed. This changes the code so that LogErr checks if failure is None and explicitly creates a Failure object.

I also updated transform_eq to be a bit more useful when debugging - it now __repr__s to have a lot more useful information.

### CODE REVIEW

File: otter/test/log/test_intents.py [hunk 1]
```diff
@@ -84,6 +84,25 @@ def test_err(self):
...
+        try:
+            raise RuntimeError('original')
+        except RuntimeError:
+            eff = err(None, "why")
+        try:
+            raise RuntimeError('performing')
+        except RuntimeError:
+            sync_perform(self.disp, eff)
+        self.log.err.assert_called_once_with(
+            transform_eq(lambda f: (f.type, str(f.value)),
```
Comment (manishtomar) [inline otter/test/log/test_intents.py]: Non-blocking thought for FYI only: You could use `CheckFailureValue` in this case but even better would be to implement that using `transform_eq` :)

**Comment by manishtomar** [comment]
LGTM once build passes :)

File: otter/test/log/test_intents.py (hunk 1 above)
Comment (radix) [inline otter/test/log/test_intents.py]: d'oh, totally forgot about that!

**Comment by radix** [comment]
jenkins please run unit tests

**Comment by radix** [comment]
jenkins please run unit tests

**Comment by radix** [comment]
ok, gonna merge this since jenkins is cool with it. Thanks for the review @manishtomar :)

**Pull request merged**